### ISSUE TITLE
[feat] `<rh-overlay>`

### ISSUE THREAD

**Issue by zeroedin** [RedHat-UX/red-hat-design-system#274]
### Description

RHDS needs a utility "overlay" component that can be used in `rh-nav`, `rh-secondary-nav` and a possible future  `rh-modal` or any other components that might need an opacity overlay layer.  The component should be of singleton scope so that if multiple components need an overlay it's not overlaid more than once.  The overlay component should be triggered with a custom `ComposedEvent`.

### Acceptance Criteria

- [x] Design done
- [ ] Development done

**Comment by bennypowers** [comment]
let's consider using [native `<dialog>`](https://web.dev/building-a-dialog-component/) under the hood for semantics and a11y

**Comment by zeroedin** [comment]
> let's consider using [native `<dialog>`](https://web.dev/building-a-dialog-component/) under the hood for semantics and a11y

Sorry, you might have to explain this more.  The feature I'm describing in itself isn't a modal / dialog but just a non-contextual visual overlay for something like a modal covering the context of the page.  If that does translate to a dialog for you help me understand.

**Comment by bennypowers** [comment]
Summarizing some discussion from various channels, `<dialog>` is PHAT, and might work semantically for various cases even things like popovers and flyouts, but we're concerned about some a11y implications, so we're holding off on adopting it.

Back to the question at hand, can we investigate if we can do this with a controller or directives (or both) rather than a new component?

**Comment by zeroedin** [comment]
For the purposes of `navigation-primary` we've decided to not abstract the overlay to its own component.  I'm going to move this issue on to diglett for review later when sidenav comes into play.

**Comment by coreyvickery** [comment]
@zeroedin Can we say the overlay should be `--rh-color-gray-95` at `50%` and call it a day?

**Comment by coreyvickery** [comment]
@zeroedin I don’t think so. I feel like overlays are reserved for mega menus, dialogs, etc.
